Tools/LyTestTools/ly_test_tools/o3de/editor_test.py

```diff
@@ -362,6 +389,28 @@ def _editor_test_data(self, request: _pytest.fixtures.FixtureRequest) -> EditorT
         else:
             editor_utils.kill_all_ly_processes(include_asset_processor=False)
 
+    @pytest.fixture(scope="function")
+    def test_launcher(self,
+                        request: _pytest.fixtures.FixtureRequest,
+                        editor: ly_test_tools.Launcher,
+                        material_editor: ly_test_tools.Launcher) -> ly_test_tools.Launcher:
+        """
+        Fixture that determines which launcher to use for the current test.
+        :request: The Pytest request object.
+        :editor: The editor Launcher object.
+        :material_editor: The material_editor Launcher object.
+        :returns: Either the editor or material_editor Launcher object.
+        """
+        return self._test_launcher(request, editor, material_editor)
+
+    def _test_launcher(self, request, editor, material_editor):
```

Comment:
Its not the best practice to add unit testing code inside the functionality, better move it away if possible. Can we do this differently?